src/lsm/segmented_array.zig

```diff
@@ -120,16 +153,55 @@ fn SegmentedArrayType(
             allocator.free(array.indexes);
         }
 
+        pub fn verify(array: Self) void {
+            for (array.nodes) |node, node_index| {
+                if (node_index < array.node_count) {
+                    // The first node_count pointers are non-null.
+                    assert(node != null);
+                } else {
+                    // The rest are non-null.
+                    assert(node == null);
+                }
+            }
+            for (array.nodes[0..array.node_count]) |_, node_index| {
+                const c = array.count(@intCast(u32, node_index));
+                // Every node is at most full.
+                assert(c <= node_capacity);
+                // Every node is at least half-full, except the last.
+                if (node_index < array.node_count - 1) {
+                    assert(c >= @divTrunc(node_capacity, 2));
+                }
+            }
+            if (Key) |K| {
+                // If Key is not null then the elements must be sorted by key_from_value (but not necessarily unique).
+                var last_key: ?K = null;
```

Comment:
Actually, would it make more sense to call this `prior_key` (or `key_prior`)? Just to disambiguate from "last in the list of elements". (I used "prior" rather than "previous" because it is shorter).

Reply:
`key_prior` is great with the qualifier always last (excuse the pun). 😛

Reply:
:+1: